cmake/onnxruntime_webassembly.cmake

```diff
@@ -204,38 +193,94 @@ else()
+    target_compile_options(nsync_cpp PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnx_proto PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(protoc PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(libprotobuf-lite PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_providers PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_optimizer PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_mlas PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_optimizer PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_graph PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_flatbuffers PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(onnxruntime_util PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(re2 PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_base PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_hash PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_raw_hash_set PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_throw_delegate PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_city PRIVATE -sMEMORY64 -Wno-experimental)
+    target_compile_options(absl_low_level_hash PRIVATE -sMEMORY64 -Wno-experimental)
+    if (onnxruntime_BUILD_UNIT_TESTS)
+      target_compile_options(gtest PRIVATE -sMEMORY64 -Wno-experimental)
+      target_compile_options(gmock PRIVATE -sMEMORY64 -Wno-experimental)
+    endif ()
+
+    target_link_options(onnxruntime_webassembly PRIVATE
+      "SHELL:-s MEMORY64=1"
+      "SHELL:-s MAXIMUM_MEMORY=17179869184"
+      --post-js "${ONNXRUNTIME_ROOT}/wasm/js_post_js_64.js"
+    )
+  else ()
+    target_link_options(onnxruntime_webassembly PRIVATE
+      "SHELL:-s MAXIMUM_MEMORY=4294967296"
+      --post-js "${ONNXRUNTIME_ROOT}/wasm/js_post_js.js"
+    )
+  endif ()
+
   target_link_options(onnxruntime_webassembly PRIVATE
     "SHELL:-s EXPORTED_RUNTIME_METHODS=${EXPORTED_RUNTIME_METHODS}"
     "SHELL:-s EXPORTED_FUNCTIONS=${EXPORTED_FUNCTIONS}"
-    "SHELL:-s MAXIMUM_MEMORY=4294967296"
     "SHELL:-s EXIT_RUNTIME=0"
     "SHELL:-s ALLOW_MEMORY_GROWTH=1"
     "SHELL:-s MODULARIZE=1"
     "SHELL:-s EXPORT_ALL=0"
     "SHELL:-s VERBOSE=0"
-    "SHELL:-s FILESYSTEM=0"
+    "SHELL:-s FILESYSTEM=1"
+    "SHELL:-s SIGNATURE_CONVERSIONS=OrtRun:_pppppppp,OrtGetTensorData:_ppppp,OrtCreateTensor:p_pppp,OrtCreateSession:pppp,OrtReleaseSession:_p,OrtGetInputOutputCount:pppp,OrtCreateSessionOptions:pp__p_ppppp,OrtAddSessionConfigEntry:pppp,OrtReleaseSessionOptions:_p,OrtAppendExecutionProvider:ppp,OrtAddSessionConfigEntry:pppp,OrtGetInputName:ppp,OrtGetOutputName:ppp,OrtCreateRunOptions:ppp_p,OrtReleaseRunOptions:pp,OrtReleaseTensor:_p,OrtFree:_p,OrtGetLastError:_pp,JsepOutput:pp_p"
```

Comment:
Since 64bit functions require BigInt as a pointer argument and may return a pointer as a BigInt, this setting creates wrappers during compile time.
In OrtRun:_ppp first value is function return value. "_" means it stays intact. "p" means pointer conversion from/to BigInt to Number. "P" makes optional conversion (checks for null/undefined)